File: scrooge-generator/src/main/resources/scalagen/writeField.scala
{{#optional}}
if ({{fieldName}}.isDefined) {
{{/optional}}
{{^optional}}
{{#nullable}}
if ({{fieldName}} ne null) {
{{/nullable}}
{{^nullable}}
if (true) {
{{/nullable}}
{{/optional}}
  val {{valueVariableName}} = {{fieldName}}{{#optional}}.get{{/optional}}
  _oprot.writeFieldBegin({{fieldConst}}{{#isEnum}}I32{{/isEnum}})
  {{>writeValue}}
  _oprot.writeFieldEnd()
}

File: scrooge-generator/src/main/resources/javagen/writeField.java
{{#optional}}if ({{fieldName}}.isDefined()) {{{/optional}}
  _oprot.writeFieldBegin({{fieldConst}}{{#isEnum}}I32{{/isEnum}});
  {{>qualifiedFieldType}} {{valueVariableName}} = {{fieldName}}{{#optional}}.get(){{/optional}};
  {{>writeValue}}
  _oprot.writeFieldEnd();
{{#optional}}}{{/optional}}
